### PR TITLE
feat: add link to log after assessment

### PR DESCRIPTION
**Pull request by cslong**
*Issue #, if available:*
https://sim.amazon.com/issues/P48091112

*Description of changes:*
Add button in flash-bar that links to assessment log file after assessment finishes (success and fail)

*Testing done:*
Tested manually on macOS and windows environment

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/porting-assistant-dotnet-ui/blob/develop/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/porting-assistant-dotnet-ui/blob/develop/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including READMEs and comments (where appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific environment

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs [hunk 1]
```diff
@@ -109,6 +110,14 @@ public void SetupConnection(bool console = false)
                     };
                 }
             });
+
+            _connection.On<string, string> ("getAssessmentLog", request => {
+                if (!DateTime.TryParse(request, out DateTime date)) {
+                  date = DateTime.Today;
+                }
+                string dateString = date.ToString("yyyyMMdd");
+                return Path.Combine(_logDirectory, $"portingAssistant-assessment-{dateString}.log");
```
Comment (flywanli) [inline packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs:119]: you do not need to move this function in backend, electron side should also be able to resolve it.

Comment (flywanli) [reply packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs:119]: The log file folder is passed as parameter from electron side

File: packages/electron/src/electron-backend.ts [hunk 2]
```diff
@@ -156,6 +156,11 @@ export const initConnection = (logger: any = console) => {
       return response;
     });
 
+    ipcMain.handle("getAssessmentLog", async (_event) => {
+      const response = await connection.send("getAssessmentLog", new Date().toISOString());
+      return response;
+    });
+
```
Comment (flywanli) [inline packages/electron/src/electron-backend.ts:163]: same here

File: packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs (hunk 1 above)
Comment (cslong) [reply packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs]: moved from backend to electron in latest changes

File: packages/electron/src/electron-backend.ts (hunk 2 above)
Comment (cslong) [reply packages/electron/src/electron-backend.ts]: removed this

File: packages/electron/src/preload.ts [hunk 3]
```diff
@@ -91,6 +91,10 @@ contextBridge.exposeInMainWorld("electron", {
   verifyUser: (profile: string) => invokeBackend("verifyProfile", profile),
   getVersion: () => invokeBackend("getVersion"),
   telemetry: (message: any) => invokeBackend("telemetry", message),
+  getAssessmentLog: () => {
+    const dateString = new Date().toISOString().slice(0,10).replace(/-/g,"");
```
Comment (marknfawaz) [inline packages/electron/src/preload.ts:95]: Are we generating this same filename somewhere else? If yes, can we reuse from that location?

**Review by marknfawaz — CHANGES_REQUESTED** [review]

File: packages/electron/src/preload.ts (hunk 3 above)
Comment (cslong) [reply packages/electron/src/preload.ts:95]: The filename is coming from the log configuration, so I don't think it's usable in this context.

https://github.com/aws/porting-assistant-dotnet-ui/blob/develop/packages/csharp/PortingAssistant/PortingAssistant.Api/Program.cs#L39

**Review by flywanli — APPROVED** [review]
LGTM

**Pull request merged**